CONTRIBUTING.md

```diff
@@ -1,81 +1,104 @@
+## 3 Version policy and development branches
 
 We basically develop in the `master` branch.
 
 1. We will keep the first version digit `0` until we have some super major changes in the project organization level.
 
-2. The second version digit will be updated when we have major updates including new functions and refactoring, and 
+2. The second version digit will be updated when we have major updates, including new functions and refactoring, and 
    their related bug fix and recipe changes.
-   This version update will be done roughly at every half year so far (but it depends on the development plan).
+   This version update will be done roughly every half year so far (but it depends on the development plan).
 
-3. The third version digit will be updated when we fix serious bugs or accumulate some minor changes including
+3. The third version digit will be updated when we fix serious bugs or accumulate some minor changes, including
    recipe related changes periodically (every two months or so).
 
-## Unit testing
+## 4 Unit testing
 
 ESPnet's testing is located under `test/`.  You can install additional packages for testing as follows:
 ``` console
 $ cd <espnet_root>
 $ pip install -e ".[test]"
 ```
 
-### Python
+### 4.1 Python
 
 Then you can run the entire test suite using [flake8](http://flake8.pycqa.org/en/latest/), [autopep8](https://github.com/hhatto/autopep8), [black](https://github.com/psf/black) and [pytest](https://docs.pytest.org/en/latest/) with [coverage](https://pytest-cov.readthedocs.io/en/latest/reporting.html) by
 ``` console
 ./ci/test_python.sh
 ```
 
-To create new test file. write functions named like `def test_yyy(...)` in files like `test_xxx.py` under `test/`.
-[Pytest](https://docs.pytest.org/en/latest/) will automatically test them. Note that, [pytest-timeouts](https://pypi.org/project/pytest-timeouts/) raises **an error when any tests exceed 2.0 sec**. To keep unittests fast, please avoid large parameters, dynamic imports, and file access. If your unittest really needs more time, you can annotate your test function with `@pytest.mark.timeout(sec)`. You can find pytest fixtures in `test/conftest.py`. [They finalize unit tests.](https://docs.pytest.org/en/latest/fixture.html#using-fixtures-from-classes-modules-or-projects)
+To create a new test file. write functions named like `def test_yyy(...)` in files like `test_xxx.py` under `test/`.
+[Pytest](https://docs.pytest.org/en/latest/) will automatically test them. Note that, [pytest-timeouts](https://pypi.org/project/pytest-timeouts/) raises **an error when any tests exceed 2.0 sec**. To keep unit tests fast, please avoid large parameters, dynamic imports, and file access. If your unit test really needs more time, you can annotate your test function with `@pytest.mark.timeout(sec)`. You can find pytest fixtures in `test/conftest.py`. [They finalize unit tests.](https://docs.pytest.org/en/latest/fixture.html#using-fixtures-from-classes-modules-or-projects)
 
```

Comment:
It may be useful to mention some general pytest options such as `--cov-report term-missing` and also give some general advices like separating big unit test (e.g.: `test_e2e_X`) into smaller ones, each one should test only methods/operations inside one file, when it's possible

Reply:
That makes sense, thank you!

Reply:
Could you check again for the pytest part. I personally do not have many experiences in pytest usage. It would be great if you could have any more suggestions.